File: water-intake-tracker/app/src/main/java/com/example/watertracker/MainActivity.java
package com.example.watertracker;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.example.watertracker.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Inflate the activity layout using the ActivityMainBinding class
        com.example.watertracker.databinding.ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Specify the top-level destinations for the app's navigation
        AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(
                R.id.navigation_home, R.id.navigation_information, R.id.navigation_settings)
                .build();

        // Get a reference to the NavController and set up the navigation
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_activity_main);
        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
        NavigationUI.setupWithNavController(binding.navView, navController);
    }
}

File: 7minWorkout/app/src/main/java/com/example/a7minworkout/MainActivity.kt
package com.example.a7minworkout

import android.app.Dialog
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import android.widget.FrameLayout
import android.widget.Toast
import com.example.a7minworkout.databinding.ActivityMainBinding
import com.example.a7minworkout.databinding.DialogCustomBackConfirmationBinding
import com.example.a7minworkout.databinding.DialogCustomBackConfirmationForMainActivityBinding

class MainActivity : AppCompatActivity() {
    private var binding:ActivityMainBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding?.root)//R.layout.activity_main isn't used here

        //val flStartButton : FrameLayout = findViewById(R.id.flStart)
        binding?.flStart?.setOnClickListener{//now we don't need to create any variables for layout items using their ids.


            val intent = Intent(this , ExcerciseActivity::class.java)
            startActivity(intent)

        }
        binding?.flBmi?.setOnClickListener{
            val intent = Intent(this , BMIActivity::class.java)
            startActivity(intent)
        }
        binding?.flHistory?.setOnClickListener{
            val intent = Intent(this , HistoryActivity::class.java)
            startActivity(intent)
        }


    }
    override fun onBackPressed() {
        customDialogForBackButton()

    }//overriding the onBackPressed functionality to modify it .
    private fun customDialogForBackButton(){
        val customDialog = Dialog(this@MainActivity)
        val dialogBinding = DialogCustomBackConfirmationForMainActivityBinding.inflate(layoutInflater)
        // inflating binding for custom xml resource file.
        customDialog.setContentView(dialogBinding.root)
        customDialog.setCanceledOnTouchOutside(false)
        dialogBinding.btnYes.setOnClickListener{
            this@MainActivity.finish()
            customDialog.dismiss()
        }
        dialogBinding.btnNahh.setOnClickListener{
            customDialog.dismiss()
        }
        customDialog.show()

    }

    override fun onDestroy() {//used to unassign viewbinding

        super.onDestroy()
        binding = null
    }


}